tests/providers/apache/spark/hooks/test_spark_sql.py

```diff
@@ -213,3 +209,29 @@ def test_spark_process_runcmd_and_fail(self, mock_popen):
                 sql, master, params, status
             ),
         )
+
+    def test_resolve_connection_yarn_default_connection(self):
+        hook = SparkSqlHook(conn_id='spark_default', sql='SELECT 1')
```

Comment:
This can be put into setup() test method, after that you will be have access through `self` to the hook and you will avoid code duplication. 
```
    def setUp(self):
            self.hook = SparkSqlHook(conn_id='spark_yarn_cluster', sql='SELECT 1')
```

Reply:
I would go even further and use https://github.com/wolever/parameterized. 
This is basically the same test case with different data `expected_spark_connection` that can be parametrized thorough @parameterized(). 
Just saying. :)

Reply:
Great idea! Yeah I am familiar with the concept as pytest has `parametrize` which I use. I will make those changes and resubmit sometime today.

Reply:
@michalslowikowski00 done!

Reply:
Cool :D